Index: web/modules/contrib/default_content_deploy/src/Importer.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- web/modules/contrib/default_content_deploy/src/Importer.php	(date 1584407602540)
+++ web/modules/contrib/default_content_deploy/src/Importer.php	(date 1584407602540)
@@ -465,9 +465,12 @@
    */
   protected function loadEntityById($entity_type, $id) {
     /** @var \Drupal\Core\Entity\Entity $entity */
-    $entityStorage = $this->entityTypeManager->getStorage($entity_type);
-    $entity = $entityStorage->load($id);
-    return $entity;
+    if (!empty($id)) {
+      $entityStorage = $this->entityTypeManager->getStorage($entity_type);
+      $entity = $entityStorage->load($id);
+      return $entity;
+    }
+    return FALSE;
   }
 
   /**
